presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -366,6 +366,17 @@ private void processNewSources()
                 return NOT_BLOCKED;
             }
 
+            // LookupJoinOperator is broken and requires finish to be called continuously
```

Comment:
Can you explain a little bit more? 
There was no contract saying `finish()` is called exactly once. `HashBuilderOperator` heavily relies on `finish` being called multiple times (up until HBO itself is finished).

As to the LJO, `finish` needs to be repeated because there's this:
```
if (!spillInProgress.isDone()) {
    return;
}
checkSuccess(spillInProgress, "spilling failed");
```

I don't know why someone put it there (although I know it was me). Apparently this can be removed. Still, HBO needs to have finish called "continuously".